fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountDomainServiceJpa.java

```diff
@@ -79,13 +83,30 @@ public SavingsAccountDomainServiceJpa(final SavingsAccountRepositoryWrapper savi
         this.context = context;
         this.depositAccountOnHoldTransactionRepository = depositAccountOnHoldTransactionRepository;
         this.businessEventNotifierService = businessEventNotifierService;
+        this.loanReadPlatformService = loanReadPlatformService;
+    }
+
+    private BigDecimal getOverdueLoanAmountForClient(SavingsAccount savingsAccount, boolean isTransferToLoanAccount) {
+        BigDecimal overdueLoanAmountForClient = BigDecimal.ZERO;
+        if (this.configurationDomainService.enforceOverdueLoansForMinBalance() && !isTransferToLoanAccount) {
+            List<LoanAccountData> loanAccountDataList = this.loanReadPlatformService
+                    .retrieveOverDueLoansForClient(savingsAccount.getClient().getId());
+            if (CollectionUtils.isNotEmpty(loanAccountDataList)) {
+                for (int i = 0; i < loanAccountDataList.size(); i++) {
+                    LoanAccountData loanAccountData = loanAccountDataList.get(0);
```

Comment:
Is this intentional passing index = 0 or it should be [i] ?

Reply:
Yes. You spot this correct. I have corrected it.